### PR TITLE
Further VLM adjustments and LLM tests parametrization pt.1

### PR DESCRIPTION
**Pull request by mzegla**
Changes:
- disabled completions echo for VLM pipeline
- enforcing both `role` and `content` existence in `message` element
- first phase of LLM tests refactor - the goal at this point is to include VLM CB servable in general CB servable testing, so we don't duplicate shared part of the functionality

### CODE REVIEW

File: src/test/llm/llmnode_test.cpp [hunk 1]
```diff
@@ -190,39 +205,59 @@ TEST_F(LLMFlowHttpTest, unaryCompletionsJson) {
...
+        ASSERT_EQ(
+            handler->dispatchToProcessor(endpointCompletions, requestBody, &response, comp, responseComponents, writer),
+            ovms::StatusCode::MEDIAPIPE_EXECUTION_ERROR);
+    }
 }
 
-TEST_F(LLMFlowHttpTest, unaryCompletionsJsonSpeculativeDecoding) {
+TEST_P(LLMFlowHttpTestParameterized, unaryCompletionsJsonSpeculativeDecoding) {
+    auto params = GetParam();
+    // TODO: In the next step we should break this suite into smaller ones, use proper configuration instead of skipping
```
Comment (dkalinowski) [inline src/test/llm/llmnode_test.cpp:243]: which jira is the next step?

File: src/test/llm/llmnode_test.cpp [hunk 2]
```diff
@@ -2841,7 +3310,14 @@ TEST_F(LLMConfigHttpTest, LLMNodeWorkspacePathToFileNotDir) {
     ASSERT_EQ(mediapipeDummy.validate(manager), StatusCode::LLM_NODE_DIRECTORY_DOES_NOT_EXIST);
 }
 
-TEST_F(LLMConfigHttpTest, LLMNodeResourceInitFailed) {
+class LLMConfigHttpTestParameterized : public ::testing::Test, public ::testing::WithParamInterface<std::tuple<std::string, ovms::StatusCode>> {
+public:
+    void SetUp() { py::initialize_interpreter(); }
```
Comment (dkalinowski) [inline src/test/llm/llmnode_test.cpp:3315]: are those the new tests that need interpreter? if so, we should modify run_unit_tests.sh to run them in separate processes

File: src/test/llm/llmnode_test.cpp (hunk 1 above)
Comment (mzegla) [reply src/test/llm/llmnode_test.cpp:243]: https://jira.devtools.intel.com/browse/CVS-164532

File: src/test/llm/llmnode_test.cpp [hunk 3]
```diff
@@ -527,17 +635,26 @@ TEST_F(LLMFlowHttpTest, unaryCompletionsJsonNFail) {
         handler->dispatchToProcessor(endpointCompletions, requestBody, &response, comp, responseComponents, writer),
         ovms::StatusCode::MEDIAPIPE_EXECUTION_ERROR);
 }
-TEST_F(LLMFlowHttpTest, unaryCompletionsJsonN) {
+
+TEST_P(LLMFlowHttpTestParameterized, unaryCompletionsJsonN) {
+    auto params = GetParam();
+    // TODO: In the next step we should break this suite into smaller ones, use proper configuration instead of skipping
+    if (params.modelName.find("vlm") != std::string::npos) {
```
[the remark this answers is not in the document]
Comment (mzegla) [reply src/test/llm/llmnode_test.cpp:642]: This is all temporary. All completions tests should not run with VLM since we decided to reject requests on completions endpoint for VLM servable. They are skipped now, but I would like them organized in smaller suits and avoid such skipping. Not sure if it's worth to refine that logic now if we want it removed soon.

**Review by dkalinowski — APPROVED** [review]

**Review by michalkulakowski — APPROVED** [review]
